src/dotnet/Telemetry.cs

```diff
@@ -9,18 +10,28 @@ namespace Microsoft.DotNet.Cli
 {
     public class Telemetry : ITelemetry
     {
+        public const double TimeoutInMilliseconds = 100;
+
         private bool _isInitialized = false;
         private TelemetryClient _client = null;
 
         private Dictionary<string, string> _commonProperties = null;
         private Dictionary<string, double> _commonMeasurements = null;
+        private Task _trackEventTask = null;
+
+        private const int ReciprocalSampleRateValue = 5;
```

Comment:
A question I have about sampling:

When we did Telemetry in the VS features I worked on, everyone used the scenario of "first the user did X, then the user did Y".  If we sample, there is no way we can tie two commands together any more.  Is that OK?  Do we care if a user first did restore and then build?  or build and then publish and then test?